### PR TITLE
Directed modularity

### PR DESCRIPTION
**Pull request by jg-you**
Related to Pull Request https://github.com/networkx/networkx/pull/1448 
This pull request implements the directed modularity as defined in E. A. Leicht, M. E. J. Newman,        "Community structure in directed networks",  Phys. Rev Lett., vol. 100, no. 11, p. 118703, 2008.
([arXiv link, see eq.3](http://arxiv.org/pdf/0709.4500v1.pdf)).

**Changes**

* `directed_modularity_matrix`
* associated documentation
* associated spectrum functions
* tests:`test_directed_modularity` (added), `test_modularity_spectrum` (updated)

I also fixed some typos and added an example in the  `modularity_matrix` function documentation.

You'll find that the definition of the modularity does not correspond exactly to the one of Leicht and Newman. NetworkX defines the element A_ij of the adjacency matrix as 1 if there is a link going from node i to node j. Leicht and Newman use the opposite definition.

Note that I followed the design of [networkx/linealg/laplacianmatrix.py](networkx/linealg/laplacianmatrix.py) and implemented the directed modularity matrix in its own function. It can be moved within `modularity_matrix` if requested.

### CODE REVIEW

**Comment by jg-you** [comment]
Sorry for the close/open spamming ~ forgot to squash and panicked. :]

**Comment by chebee7i** [comment]
It looks like you'll have to update some of the unit tests so that they require scipy, or something like that. Also, some tests were failing due to print statements.

**Comment by chebee7i** [comment]
Also, be sure to update the release notes. And consider adding yourself, a research group, or funding sources to `credits.rst`, if desired.

**Comment by jg-you** [comment]
I was not aware that travis-ci ran tests on documentated examples. This seems to cause all the failures. Also, is [doc/source/reference/news.rst](doc/source/reference/news.rst) the proper file for change logs?

**Comment by jg-you** [comment]
Welp, there's minor merge conflict with credits.rst

File: networkx/linalg/modularitymatrix.py [hunk 1]
```diff
@@ -61,6 +74,73 @@ def modularity_matrix(G, nodelist=None):
...
+    --------
+    >>> import networkx as nx
+    >>> G = nx.DiGraph()
+    >>> G.add_edges_from(((1,2), (1,3), (3,1), (3,2), (3,5), (4,5), (4,6),
+    ...                   (5,4), (5,6), (6,4)))
+    >>> B = nx.directed_modularity_matrix(G)
+
+
+    Notes
+    _____
```
Comment (chebee7i) [inline networkx/linalg/modularitymatrix.py]: Are these hyphens?! I can't tell if my reader is deceiving me.

Comment (jg-you) [inline networkx/linalg/modularitymatrix.py]: Yes they are D:.

**Comment by jg-you** [comment]
Sorry, i was trying to squash and I think you merged at one point.

**Comment by chebee7i** [comment]
No worries, still looks clean. If I was doing something with git that isn't recommended, let me know :)

**Comment by jg-you** [comment]
I think I'm at fault haha! Not supposed to squash something that is shared with others. 
I'll just stop touching it, sorry for the dirty history,

**Pull request merged**